eng/Version.Details.xml

```diff
@@ -9,384 +9,464 @@
     </Dependency>
-    <Dependency Name="Microsoft.EntityFrameworkCore.SqlServer" Version="3.0.0-preview.19080.1">
+    <Dependency Name="Microsoft.EntityFrameworkCore.SqlServer" Version="3.0.0-preview4.19124.2">
+      <CoherentParentDependency>Microsoft.NETCore.App</CoherentParentDependency>
       <Uri>https://github.com/aspnet/EntityFrameworkCore</Uri>
-      <Sha>df4350283ae7ff0c67e08af16c0f00fdbd5e22ac</Sha>
+      <Sha>d1a2d66354d42ff951048124252f81d051736ffe</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.EntityFrameworkCore.Tools" Version="3.0.0-preview.19080.1">
+    <Dependency Name="Microsoft.EntityFrameworkCore.Tools" Version="3.0.0-preview4.19124.2">
+      <CoherentParentDependency>Microsoft.NETCore.App</CoherentParentDependency>
       <Uri>https://github.com/aspnet/EntityFrameworkCore</Uri>
-      <Sha>df4350283ae7ff0c67e08af16c0f00fdbd5e22ac</Sha>
+      <Sha>d1a2d66354d42ff951048124252f81d051736ffe</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.EntityFrameworkCore" Version="3.0.0-preview.19080.1">
+    <Dependency Name="Microsoft.EntityFrameworkCore" Version="3.0.0-preview4.19124.2">
+      <CoherentParentDependency>Microsoft.NETCore.App</CoherentParentDependency>
       <Uri>https://github.com/aspnet/EntityFrameworkCore</Uri>
-      <Sha>df4350283ae7ff0c67e08af16c0f00fdbd5e22ac</Sha>
+      <Sha>d1a2d66354d42ff951048124252f81d051736ffe</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.AspNetCore.Analyzer.Testing" Version="3.0.0-preview3.19115.4">
+    <Dependency Name="Microsoft.AspNetCore.Analyzer.Testing" Version="3.0.0-preview4.19122.13">
+      <CoherentParentDependency>Microsoft.NETCore.App</CoherentParentDependency>
       <Uri>https://github.com/aspnet/Extensions</Uri>
-      <Sha>321f30556d63ec8af6eb0bc43d534b6b2e5458a0</Sha>
+      <Sha>f263e24ff720381ba0c17f10ce0b4d1725e1c062</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.AspNetCore.BenchmarkRunner.Sources" Version="3.0.0-preview3.19115.4">
+    <Dependency Name="Microsoft.AspNetCore.BenchmarkRunner.Sources" Version="3.0.0-preview4.19122.13">
       <Uri>https://github.com/aspnet/Extensions</Uri>
-      <Sha>321f30556d63ec8af6eb0bc43d534b6b2e5458a0</Sha>
+      <Sha>f263e24ff720381ba0c17f10ce0b4d1725e1c062</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.Extensions.ActivatorUtilities.Sources" Version="3.0.0-preview3.19115.4">
+    <Dependency Name="Microsoft.Extensions.ActivatorUtilities.Sources" Version="3.0.0-preview4.19122.13">
       <Uri>https://github.com/aspnet/Extensions</Uri>
-      <Sha>321f30556d63ec8af6eb0bc43d534b6b2e5458a0</Sha>
+      <Sha>f263e24ff720381ba0c17f10ce0b4d1725e1c062</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.Extensions.Caching.Abstractions" Version="3.0.0-preview3.19115.4">
+    <Dependency Name="Microsoft.Extensions.Caching.Abstractions" Version="3.0.0-preview4.19122.13">
+      <CoherentParentDependency>Microsoft.NETCore.App</CoherentParentDependency>
```

Comment:
This presents a new challenge I didn't think much about till now.

If AspNetCore-Tooling depends on Extensions build N and EntityFrameworkCore depends on build N+1, which one should darc choose? N or N+1? What if there is no build of AspNetCore-Tooling and EFCore that both depend on the same version of Extensions?

cc @mmitche

Reply:
@natemcmaster if we need those two to depend on a coherent set of aspnet/Extensions packages, perhaps we need to be able to specify `<CoherentParentDependency>Microsoft.NETCore.App;Microsoft.Extensions.Configuration</CoherentParentDependency>` or some such?

/cc @mmitche @jcagme

Reply:
The bot automerges EF dependencies without interaction; i.e. no review required (e.g. https://github.com/aspnet/EntityFrameworkCore/pull/14799); Extensions always requires an approval to merge, so its probably why EF is always getting ahead with dependencies?

Reply:
i.e. either allow the bot to automerge Extensions (so they are both always on latest fairly close to each other); or require approval for EF so the merges can be coordinated

Reply:
@benaadams I'm playing with a new version of the tool we hope the bot will use eventually. If the tool can't avoid inchorent dependencies, the bot will be worse.

Reply:
> This presents a new challenge I didn't think much about till now.
> 
> If AspNetCore-Tooling depends on Extensions build N and EntityFrameworkCore depends on build N+1, which one should darc choose? N or N+1? What if there is no build of AspNetCore-Tooling and EFCore that both depend on the same version of Extensions?
> 
> cc @mmitche

The rule right now is that we choose a build that has the minimal graph depth.  So if you say that some corefx dependency should be coherent with M.N.A, then it will find the corefx build producing the dependency with minimal depth to the current M.N.A version.  Since, for instance, core-setup depeneds on coreclr which depends on corefx, that path to a corefx build will NOT get chosen because it is longer than M.N.A. to corefx.  If no coherent build is possible (e.g. your dependency is on a new corefx binary who's build has not yet been pulled into core-setup) then the tool will error, as no coherent build is possible.  In this case, you can either remove the attribute, or set the dependency to pinned.

Let's whiteboard this in the morning so I can get a better picture of what needs to happen.